plugins/woocommerce/src/Internal/EmailEditor/WCTransactionalEmails/WCTransactionalEmailPostsGenerator.php

```diff
@@ -0,0 +1,273 @@
+	}
+
+	/**
+	 * Initialize the email template generator.
+	 *
+	 * This function initializes the email template generator by loading the default templates
+	 * and generating initial email templates if needed.
+	 *
+	 * @internal
+	 */
+	public function initialize() {
+		if ( Constants::get_constant( 'WC_VERSION' ) === get_transient( $this->transient_name ) ) {
+			// if templates are already generated, we don't need to run this function again.
+			return true;
+		}
+
+		$this->init_default_transactional_emails();
+		$this->generate_initial_email_templates();
+	}
+
+	/**
+	 * Initialize the default WooCommerce Transactional Emails.
+	 *
+	 * This function initializes the default templates for the core transactional emails.
+	 * It fetches all the emails from WooCommerce and filters them to include only the core transactional emails.
+	 */
+	public function init_default_transactional_emails() {
+		$core_transactional_emails = WCTransactionalEmails::get_transactional_emails();
+
+		$wc_emails = \WC_Emails::instance();
+		/**
+		 * WooCommerce Transactional Emails instance.
+		 *
+		 * @var \WC_Email[]
+		 */
+		$email_types = $wc_emails->get_emails();
+
+		// Filter the emails to include only the core transactional emails.
+		$email_types = array_filter(
+			$email_types,
+			function ( $email ) use ( $core_transactional_emails ) {
+				return in_array( $email->id, $core_transactional_emails, true );
+			}
+		);
+
+		$this->default_templates = array_reduce(
+			$email_types,
+			function ( $acc, $email ) {
+				$acc[ $email->id ] = $email;
+				return $acc;
+			},
+			array()
+		);
+	}
+
+	/**
+	 * Get the email template for the given email.
+	 *
+	 * Looks for the initial email block content in plugins/woocommerce/templates/emails/block.
+	 *
+	 * @param \WC_Email $email The email object.
+	 * @return string The email template.
+	 */
+	public function get_email_template( $email ) {
+		$template_name = str_replace( 'plain', 'block', $email->template_plain );
+
+		try {
+			$template_html = wc_get_template_html(
+				$template_name,
+				array()
+			);
+		} catch ( \Exception $e ) {
+			$template_html = '';
+		}
+
+		// wc_get_template_html does not throw an error when the template is not found.
+		// We need to check if the template is not found by checking the template_html content.
+		$has_template_error =
+			StringUtil::contains( $template_html, 'No such file or directory', false ) ||
+			StringUtil::contains( $template_html, 'Failed to open stream', false ) ||
+			StringUtil::contains( $template_html, 'Warning: include', false );
+
+		if ( is_wp_error( $template_html ) || empty( $template_html ) || $has_template_error ) {
+			$default_template_name = 'emails/block/default-block-content.php';
+			$template_html         = wc_get_template_html(
+				$default_template_name,
+				array(
+					'order'         => $email->object,
+					'sent_to_admin' => true,
+					'plain_text'    => false,
+					'email'         => $email,
+				)
```

Comment:
I missed this place before. I assume we can use an empty array here the same as the above.